### DISCUSSION TITLE
Is there a way to do transient processes (like power up or load changes)?

### DISCUSSION THREAD

**Discussion by stockh0lm** [oemof/tespy#542] [Q&A]
Is there a way to model or estimate transient processes in TesPy?

I am thinking of the power-up process of a heat pump or its transition from design to some off-design point and back.
E.g, the pressure in the compressors would change over time, the mass flow through the pipes and heat exchangers would shift in sync with the pumps, and that process would follow physical laws. 

One workaround for a full-blown solution (minus oscillations and other fancy shit) could be the calculation of several off-design points in between, as long as that is possible. Some states after power-up don't resemble a working heat pump meaningfully.

**Comment by fwitte** [comment]
Hi @stockh0lm,

unfortunately ramp up and ramp down simulation or some kind of rapid change in the operation process of a plant cannot be simulated very well using tespy since it is a steady state approach. Time step length has to be sufficient and the operating point has to be stable to some extent in order to allow such assumptions. That said: Your workaround is a good approach if you can ignore everything between two steady states. I think it depends on the context of what you want to achieve with your model, can you provide some information?

There was an open source modelica library once (discontinued unfortunately, and one of the reasons why tespy came into existence), which looks like something that could handle transient operation, i.e. http://thermocycle.squoilin.eu/download/thermocycle-library/.

Best

Francesco

**Comment by stockh0lm** [comment]
Thanks for your reply, @fwitte! 

The context of what I want to do is the discontinuous and batch-like operation of equipment. It is well known that, e.g., heat pumps that operate not continuously but in an on-off/pulsed fashion have a lower overall COP. An essential part is that it takes the compressors several seconds to build up the pressure while the machine operates at full power and produces no heat yet. 

For that reason, if simulating batch operation efficiency, it would be prudent to consider ramp-up and ramp-down, too, especially if doing frequent on-off operations.

Pumps suffer from similar issues, but less so because a tiny mass can be moved just like the nominal value—there is no power spike at which things suddenly start working.
